src/ua_types_encoding_binary.c

```diff
@@ -778,8 +786,15 @@ ExtensionObject_decodeBinary(UA_ExtensionObject *dst, const UA_DataType *_) {
         /* helping clang analyzer, typeId is numeric */
         UA_assert(typeId.identifier.byteString.data == NULL);
         UA_assert(typeId.identifier.string.data == NULL);
-        typeId.identifier.numeric -= UA_ENCODINGOFFSET_BINARY;
-        findDataType(&typeId, &type);
+        size_t i;
+        for (i = 0; i<UA_TYPES_COUNT; i++) {
```

Comment:
FindDataTypes only applies when we get a type in binary encoding.
Otherwise, we already have the pointer to the type, don't we. :smirk: